plugins/linux-v4l2/v4l2-output.c

```diff
@@ -220,18 +209,63 @@ static bool virtualcam_start(void *data)
 			     "v4l2-output: A format truncation may have occurred."
 			     " This can be ignored since it is quite improbable.");
 
-		if (try_connect(vcam, device)) {
-			success = true;
-			break;
+		int fd = open(device, O_RDWR);
+		if (fd < 0)
+			continue;
+
+		if (ioctl(fd, VIDIOC_QUERYCAP, &v4l2cap) < 0)
```

Comment:
leaving these checks that must jump inside try_configure to avoid these goto's seems a bit cleaner.